core/src/subsystems/tank_drive.cpp

```diff
@@ -88,22 +89,36 @@ bool TankDrive::drive_forward(double inches, double speed, double correction, di
   // Generate a point X inches forward of the current position, on first startup
   if (!func_initialized)
   {
-    saved_pos = odometry->get_position();
-    drive_pid.reset();
+    position_t cur_pos = odometry->get_position();
+
+    // forwards is positive Y axis, backwards is negative
+    if (dir == directionType::rev)
+      inches = -fabs(inches);
+    else
+      inches = fabs(inches);
 
     // Use vector math to get an X and Y
-    Vector2D current_pos({saved_pos.x , saved_pos.y});
-    Vector2D delta_pos(deg2rad(saved_pos.rot), inches);
-    Vector2D setpt_vec = current_pos + delta_pos;
+    Vector2D cur_pos_vec({cur_pos.x , cur_pos.y});
+    Vector2D delta_pos_vec(deg2rad(cur_pos.rot), inches);
+    Vector2D setpt_vec = cur_pos_vec + delta_pos_vec;
 
     // Save the new X and Y values
     pos_setpt = {.x=setpt_vec.get_x(), .y=setpt_vec.get_y()};
 
```

Comment:
It looks like `func_initialized = true` got removed from here, is this running everything in the if block every time or am I missing how it avoids that?